packages/ui/src/app/pages/WorkingGroups/MyApplications.tsx

```diff
@@ -18,20 +19,35 @@ export const MyApplications = () => {
     ],
     []
   )
-  const { applications } = useMyApplications()
+  const { applications, isLoading } = useMyApplications()
+  const isPending = (a: WorkingGroupApplication) => a.status == 'ApplicationStatusPending'
+  const currentApplications = useMemo(() => applications?.filter(isPending), [applications])
+  const pastApplications = useMemo(() => applications?.filter((a) => !isPending(a)), [applications])
```

Comment:
Keep in mind that React checks the dependency list by reference comparison which will not work with complex objects. So passing an object/array (even with the same values) will cause this to trigger re-render. You can use primitives to prevent that.

For now, we don't have to change that. The `useMyApplications()` uses memo internally.

I'd suspect, however, that the `useMemo()` dependency list inside `useMyApplications()` might be not ideal there.